feat(mep): Generic metric dataset

wmak opened the pull request:
- This properly switches the metric query builder over to the generic metrics dataset
- This currently means that local metrics data will no longer work

wmak committed:
feat(mep): Change Metric granularities to enum instead

- This changes the metric granularities to an enum instead of the direct
  raw values
  - Only changing in tables, cause I don't see how we'll do this with
    timeseries

wmak committed:
Merge branch 'master' into wmak/feat/generic-metric-dataset

evanh reviewed (approved):
Looks good to me.

silent1mezzo commented on src/sentry/search/events/builder.py:
```diff
@@ -1602,10 +1602,17 @@ def __init__(
         self.dry_run = dry_run
         super().__init__(
             # Dataset is always Metrics
```
We should remove this comment too.

wmak replied:
👍 updated in latest commit

wmak committed:
fix: Handle the sessions dataset

- This makes a bit of a TODO to come back and rename the
  Dataset.Metrics to Dataset.PerformanceMetrics
- Also defaulting to Metrics for now as the default dataset to get tests
  passing, but will likely make it PerformanceMetrics instead in the
  long run
- This also changes the hack to only happen when on the
  PerformanceMetrics dataset

wmak committed:
fix: get cm

andriisoldatenko commented on src/sentry/search/events/builder.py:
```diff
@@ -1600,12 +1603,20 @@ def __init__(
             *args,
             **kwargs,
         )
+        if dataset is Dataset.PerformanceMetrics:
```
nit: python `is` keyword is used to test object identity. I think better to rewrite as:
```suggestion
        if isinstance(dataset, Dataset.PerformanceMetrics)
```

wmak replied:
Are you sure that applies here? Dataset is a class of Enums

andriisoldatenko replied:
nvm yeah you are right, you want to check against `Dataset.PerformanceMetrics` which is member of Enum.

ahmedetefy commented on src/sentry/search/events/builder.py:
```diff
@@ -1600,12 +1603,20 @@ def __init__(
         super().__init__(
-            # Dataset is always Metrics
-            Dataset.Metrics,
+            Dataset.Metrics if dataset is None else dataset,
```
Why are you defaulting to the metrics dataset here ?

wmak replied:
Its an allowed option cause incident code is using the MetricQueryBuilder too,
And I defaulted to metrics to reduce the number of changes I need to make to incident code/tests, I'll comment this in the code too but longterm I plan to change this so the default is PerformanceMetrics instead.

ahmedetefy reviewed (approved):
lgtm

wmak committed:
fix: move generic functions to their own dictionary so i dont break
metrics stuff

- Also commenting that Metrics won't be the forever default

wmak committed:
fix: get cm

wmak committed:
fix: wrong dataset for meta

wedamija commented on src/sentry/search/events/builder.py:
```diff
@@ -1600,12 +1603,22 @@ def __init__(
             *args,
             **kwargs,
         )
+        if dataset is Dataset.PerformanceMetrics:
+            # Hack while granularity isn't being handled by snuba
+            granularity = {
+                60: 1,
+                3600: 2,
+                86400: 3,
+            }[self.granularity.granularity]
+            self.where.append(Condition(Column("granularity"), Op.EQ, granularity))
```
Are these the only 3 granularities we allow here?

onewland replied:
yeah, we dropped 10s (on the Snuba side) for performance metrics because performance has no use case for that and it creates a ton of data.
